lib/github/oauth.js

```diff
@@ -22,15 +23,29 @@ module.exports = (robot) => {
   robot.route().get('/github/oauth/callback', async (req, res) => {
     // @todo: handle condition where state is expired or has been tampered with
     const state = await SignedParams.load(req.query.state);
-    const command = await PendingCommand.find(state.trigger_id);
 
-    req.log({ state, command }, 'Restoring state to authenticate user');
-
-    if (!command) {
-      // FIXME: handle issue with expired command
-      throw new Error();
+    let command;
+    let teamSlackId;
+    let channelSlackId;
+    let userSlackId;
+    if (state.trigger_id) {
+      command = await PendingCommand.find(state.trigger_id);
+
+      if (!command) {
+        // FIXME: handle issue with expired command
+        throw new Error();
+      }
+      teamSlackId = command.team_id;
+      channelSlackId = command.channel_id;
+      userSlackId = command.user_id;
+    } else {
+      teamSlackId = state.slackEvent.team_id;
+      userSlackId = state.slackEvent.event.user;
+      channelSlackId = state.slackEvent.event.channel;
```

Comment:
Yeah I agree this would be better. I wasn't sure whether this would break `PendingCommand` in some way, because it relies on `trigger_id`? Also: would this get addressed as part of the general command refactor?

I'd prefer to open an issue for this, because I want to get back to hack week 🙈 haha